src/espresso.js

```diff
@@ -102,7 +103,7 @@ export default class Espresso {
   }
 
   async run() {
-    await this.server.start();
+    await this.server.start(this.port);
```

Comment:
And then instead of passing `this.port` here, the `Server` instance would already know what port to start on.

Reply:
Updated. Check out changes